### ISSUE TITLE
KeyError: '/Resources'

### ISSUE THREAD

**Issue by DL6ER** [py-pdf/PyPDF2#1272]
See https://github.com/py-pdf/PyPDF2/issues/1269 for further details.

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
Linux-5.4.0-122-generic-x86_64-with-glibc2.29

$ python -c "import PyPDF2;print(PyPDF2.__version__)"
2.10.3
```
## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from PyPDF2 import PdfReader
with open("TelemetryTX_EM.pdf", "rb") as f:
    reader = PdfReader(f, strict=False)
    full_content = " ".join([page.extract_text() for page in reader.pages])
```

PDF used above: [main.pdf](https://github.com/eagletrt/wiki/blob/0f3f16309604f665a47595c890d15af1b3aec6d6/fenice-telemetry-tx/PCB%20Outputs/Pdf/Edge%20Mount%20SMA/TelemetryTX_EM.pdf)

## Traceback

This is the complete Traceback I see:

```
Traceback (most recent call last):
  File "test4.py", line 4, in <module>
    content = " ".join([page.extract_text() for page in reader.pages])
  File "test4.py", line 4, in <listcomp>
    content = " ".join([page.extract_text() for page in reader.pages])
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/_page.py", line 1510, in extract_text
    return self._extract_text(
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/_page.py", line 1143, in _extract_text
    resources_dict = cast(DictionaryObject, obj["/Resources"])
  File "/usr/local/lib/python3.8/dist-packages/PyPDF2/generic/_data_structures.py", line 150, in __getitem__
    return dict.__getitem__(self, key).get_object()
KeyError: '/Resources'
```

**Comment by DL6ER** [comment]
Related to https://github.com/py-pdf/PyPDF2/issues/270

**Comment by MasterOdin** [comment]
Attaching the linked PDF into this issue:
[TelemetryTX_EM.pdf](https://github.com/py-pdf/PyPDF2/files/9428434/TelemetryTX_EM.pdf)

**Comment by DL6ER** [comment]
Just for the record, leaving more potential testing files here:

- [example.pdf](https://github.com/raspbian-packages/firefox-esr/blob/6692a5cacdd89e4a9040fadc5a58803b49de22c4/testing/web-platform/tests/loading/early-hints/resources/example.pdf)
- [Elementary Differential Equations and Boundary Value Problems (William E. Boyce, Richard C. DiPrima etc.) (z-lib.org).pdf](https://github.com/WilliamYKZ/Math441/blob/d5ad7778a6777d768612e6cfaf2de94f6733af6f/TextBook/Elementary%20Differential%20Equations%20and%20Boundary%20Value%20Problems%20%28William%20E.%20Boyce%2C%20Richard%20C.%20DiPrima%20etc.%29%20%28z-lib.org%29.pdf)
- [Hacking_APIs.pdf](https://github.com/richard1230/LearningCode/blob/2bd5980d1dbfec81d071090c8da1d8752818cdcf/Bugbounty/Writeup/Access_control_import_%E9%87%8D%E7%82%B9/API_Hacking_import_%E9%87%8D%E7%82%B9_%E6%9C%89%E6%80%BB%E7%BB%93/Hacking_APIs.pdf)
- [公积金.pdf](https://github.com/guangde/guangde.github.io/blob/931eae9d44e1c48cffe2db237a1ae59e8cc839f2/%E5%85%AC%E7%A7%AF%E9%87%91.pdf)
- [demo.pdf](https://github.com/ng-alain/archive-docs/blob/c2e1cc9e4bea4a5dd4edf6fa43d76ab40aef19be/version/13.x/assets/demo.pdf)
- [1661669306_journal-05012021083314.pdf](https://github.com/krismandev/sibiodiversity/blob/6267334a868cd645119a86ccee608c07c23dea20/public/gallery/1661669306_journal-05012021083314.pdf)
- [ee09_210604.pdf](https://github.com/Pawel123j/zawodowe/blob/84c953e0eb6b6e6f1d2a7bbf567f03448130d648/EE.09-04-21.06-SG/sposub1/ee09_210604.pdf)
- [test1.pdf](https://github.com/guozhenzhengit/test/blob/e04ae36937d6719c51dbadcbec780a04fb4d9833/test1.pdf)
- [Taller1.pdf](https://github.com/juanchitololxd/Apuntes/blob/97233787cc74fb44b0c60fb957b36b5edaf11932/RECO/OTROS/Taller1.pdf)
- [Preparing-for-the-ACT_2020-2021.pdf](https://github.com/AbdullahAmeen/AbdullahAmeen.github.io/blob/db73b2451ed79e3a5f74bfee3bc8b2a1a8bef673/FreeMathResources/contents/Preparing-for-the-ACT_2020-2021.pdf)
- [BireyselRapor.pdf](https://github.com/afur01/BSVG-Yaz-D-nemi-Yap-lanlar/blob/7a5177387c666627ad279f6a02ec503419fe10fa/BireyselRapor.pdf)
- [EE.09-02-20.01-SG.pdf](https://github.com/Pawel123j/zawodowe/blob/1f2ef25fee09ee892cf28999c15f816cea6bde98/EE.09-02-20.01-SG/EE.09-02-20.01-SG.pdf)
- [Think like a programmer.pdf](https://github.com/theritiktiwari/dev-resources/blob/be2ee0986bd8042394b454c29fcd0cf9c4f4ec28/Other/Think%20like%20a%20programmer.pdf)
- [pdfdata_-_Med_Ebook (56).pdf](https://github.com/Lexicanium/Lexicanium/blob/9eba11bef30abeeff56e5940caf756c52319307a/data/named/100/pdfdata_-_Med_Ebook%20%2856%29.pdf)
- [TelemetryTX_PM.pdf](https://github.com/eagletrt/wiki/blob/f87e45c7f9d91f525c6df395422019193856869f/fenice-telemetry-tx/PCB%20Outputs/Pdf/Panel%20Mount%20SMA/TelemetryTX_PM.pdf)
- [Reincidência Criminal no Brasil - Relatório de Pesquisa - 2015.pdf](https://github.com/micaelsev/projeto-arp/blob/11d0913b363e3640d87c274aa280eb2834a7a964/documentos/artigos-e-estatisticas/estatisticas-criminais-e-informa%C3%A7%C3%B5es-gerais/Reincid%C3%AAncia%20Criminal%20no%20Brasil%20-%20Relat%C3%B3rio%20de%20Pesquisa%20-%202015.pdf)
- [gongjijin.pdf](https://github.com/guangde/guangde.github.io/blob/d15720cb26127d689015f2ff11506d37886de8bb/gongjijin.pdf)

**Comment by MartinThoma** [comment]
@DL6ER This issue should have been fixed via #1276 in `PyPDF2==2.10.4`. Could you please check / confirm if that is the case?

If not, I will re-open this issue.

**Comment by DL6ER** [comment]
Yes, it looks like `2.10.4` works well for these files. Thanks!
